ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -131,7 +134,12 @@ public void start() throws StoreException {
             sessionId, storeDescriptor.getIncarnationId());
         compactor.initialize(index);
         metrics.initializeIndexGauges(index, capacityInBytes);
-        blobStoreStats = new BlobStoreStats(index, time, diskIOScheduler);
+        long logSegmentForecastOffsetMs =
+            config.storeDeletedMessageRetentionDays * Time.MinsPerDay * Time.SecsPerMin * Time.MsPerSec;
+        blobStoreStats = new BlobStoreStats(index, config.storeStatsBucketCount, config.storeStatsBucketSpanInMinutes,
+            logSegmentForecastOffsetMs, config.storeStatsQueueProcessorPeriodInMinutes, time, longLiveTaskScheduler,
+            taskScheduler, diskIOScheduler, metrics);
+        blobStoreStats.start();
```

Comment:
Implementing `Closable` is an option. However, you would still need some sort of function to restart the scheduled tasks when reloading the `BlobStore`. Unless we re-instantiate `BlobStoreStats` inside BlobStore's `start`.